test/unit_test/static/prime_test.cpp

```diff
@@ -0,0 +1,62 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <units/bits/prime.h>
+
+#include <type_traits>
+#include <utility>
+
+namespace units::detail
```

Comment:
```suggestion
using namespace units::detail;

namespace
```

Reply:
Learning opportunity for me here: what's the advantage of doing it this way?  I've seen both ways (top-level anonymous namespace; or, putting the tests in the same namepsace).  Are there advantages/disadvantages for each?

Reply:
I suggested it because that's just how other tests do it. As it stands, my suggestion might not be enough since the tests could be using names in `::units`.
```suggestion
using namespace units;
using namespace units::detail;

namespace
```

> Are there advantages/disadvantages for each?
Probably. I don't remember. You certainly have more opportunities to mess up if you reopen a namespace you don't own rather than using using declarations or using directives. See https://quuxplusone.github.io/blog/2021/10/27/dont-reopen-namespace-std/#don-t-do-this-instead-prefer-to for, which is related to `std` and thus probably more dangerous than reopening `units::detail`.

Reply:
Putting all the tests in the same named namespace may end up as an ODR violation if someone will copy-paste some test fixtures or just create a type with the same name. Anonymous namespace guarantees internal linkage so no ODR violations between TUs possible.

This is why I use `detail` subnamespace only in the header files where the usage of an anonymous one is a problem. For implementation details of `*.cpp` file I always prefer anonymous namespaces.